### PR TITLE
Balancer internals: use strset implicitly in backends.Update

### PR DESCRIPTION
**Pull request by mxmCherry**
As discussed in https://github.com/bsm/grpclb/pull/7

### CODE REVIEW

File: balancer/backends.go [hunk 1]
```diff
@@ -157,5 +145,14 @@ func (b *backends) updateBackendScores() error {
 		}
 		succeeded = append(succeeded, addr)
 	}
-	return b.Update(toStrset(succeeded))
+	return b.Update(succeeded)
+}
+
+// --------------------------------------------------------------------
+
+type strset []string
```
Comment (dim) [inline balancer/backends.go]: Do we need strset at all?

Comment (mxmCherry) [inline balancer/backends.go]: Well, no, it can be done with smth like `stringsContain(addrs, addr)` instead.

Comment (dim) [inline balancer/backends.go]: that's what I meant

File: balancer/backends.go [hunk 2]
```diff
@@ -157,5 +145,14 @@ func (b *backends) updateBackendScores() error {
...
+
+func (s strset) Contains(v string) bool {
+	pos := sort.SearchStrings(s, v)
```
Comment (dim) [inline balancer/backends.go]: just inline these two lines, no need for strset or even an extra func, just `if pos := ... ; pos < len(s) && s[pos] == v {...}`

**Pull request merged**